x/auction/keeper/msg_server_submit_open_bid.go

```diff
@@ -0,0 +1,77 @@
+package keeper
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/spf13/cast"
+
+	"github.com/Stride-Labs/stride/v5/x/auction/types"
+)
+
+// Get the name of the target pool, if the end-block is already before the current blockchain height, then we can start a new auction
+
+func (k msgServer) SubmitOpenBid(goCtx context.Context, msg *types.MsgSubmitOpenBid) (*types.MsgSubmitOpenBidResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	now := cast.ToUint64(ctx.BlockHeight())
+
+	ctx.Logger().Info(fmt.Sprintf("[auction]"))
+	ctx.Logger().Info(fmt.Sprintf("[auction] Open bid incoming %s at block %d", msg.GetBid(), now))
+
+	auctionPools := k.Keeper.GetAllAuctionPool(ctx)
+	for _, pool := range auctionPools {
```

Comment:
We should remove this for loop and use GetAuctionPool function instead to get auction pool by id.